src/s2_maybe.erl

```diff
@@ -51,6 +52,19 @@ do_test() ->
        ]).
 -endif.
 
+-spec fmap(fun((A) -> B), maybe(A, _)) -> maybe(B, _).
+%%@doc fmap(F, Maybe) is the result of mapping F over Maybe.
+fmap(F, {ok, Value})     when is_function(F, 1) -> {ok, F(Value)};
+fmap(F, {error, Reason}) when is_function(F, 1) -> {error, Reason}.
+
+-ifdef(TEST).
+fmap_test() ->
+  {ok, 2}               = fmap(fun(X)  -> X + 1            end, {ok, 1}),
+  {ok, {ok, 2}}         = fmap(fun(X)  -> {ok, X + 1}      end, {ok, 1}),
```

Comment:
How about putting this simple test here as well so non-haskell people get a better idea of what's going on.

```erl
{error, reason} = fmap(fun(X) -> X + 1 end, {error, reason})
```

Reply:
good idea!